go/runtime/host/sgx/epid.go

```diff
@@ -124,22 +124,24 @@ func (ep *teeStateEPID) update(
 		RuntimeID:                  ep.runtimeID,
 		Quote:                      quote,
 		Nonce:                      nonce,
+		EarlyTCBUpdate:             true,
 		MinTCBEvaluationDataNumber: quotePolicy.MinTCBEvaluationDataNumber,
 	}
 
+	// First try with early updating. If that fails, fall back to normal.
 	avrBundle, err := iasClient.VerifyEvidence(ctx, &evidence)
 	if err != nil {
-		return nil, fmt.Errorf("error while verifying attestation evidence: %w", err)
+		return nil, fmt.Errorf("error while verifying attestation evidence with early update: %w", err)
 	}
 
 	// Decode the AVR so we can do further checks.
 	avr, decErr := cmnIAS.UnsafeDecodeAVR(avrBundle.Body)
-	if decErr == nil && avr.TCBEvaluationDataNumber < quotePolicy.MinTCBEvaluationDataNumber {
-		// Retry again with early updating.
-		evidence.EarlyTCBUpdate = true
+	if decErr == nil && avr.ISVEnclaveQuoteStatus != cmnIAS.QuoteOK && avr.ISVEnclaveQuoteStatus != cmnIAS.QuoteSwHardeningNeeded {
```

Comment:
Is the last part really needed? It looks like this status is not returned anymore in v5.

Reply:
Hm where did you see that? That is not the case AFAIK.

Reply:
p19 in https://api.trustedservices.intel.com/documents/IAS-API-Spec-rev-5.0.pdf

Reply:
This is describing an old version (v3) of the API?

Reply:
You are right :facepalm: